### ISSUE TITLE
0.12.4 breaks custom text inputs field length

### ISSUE THREAD

**Issue by Orasporas** [bokeh/bokeh#5749]
So before 
![screenshot from 2017-01-14 21-27-30](https://cloud.githubusercontent.com/assets/23524922/22035451/bccb5c76-dcf8-11e6-853b-5649b6ee19fc.png)
after upgrading to 0.120.4 
![screenshot from 2017-01-14 21-33-59](https://cloud.githubusercontent.com/assets/23524922/22035487/d025d454-dcf8-11e6-9915-e64e9a600077.png)
I think it's because of this "BokehJS has migrated to using .tsx templates. Provisional support for including .eco templates has been removed from both BokehJS and the Bokeh Python library.", but i am not sure and out of ideas what to do next, code here https://github.com/Orasporas/zintistyrimas/blob/master/mainveik.py (mainveik.py).

**Comment by bryevdv** [comment]
ping @mattpap

**Comment by Orasporas** [comment]
any solution?

**Comment by bryevdv** [comment]
@Orasporas I would like to look at this soon, however I did not realized that your linked code was over 4000 lines long. Can you please condense this to a minimal test case that demonstrates the problem?

**Comment by bryevdv** [comment]
ping @Orasporas will you be able to provide a minimal test case that demonstrates the problem?

**Comment by Orasporas** [comment]
import jinja2
from bokeh.models.widgets import TextInput, Paragraph, Div, Select
from bokeh.io import curdoc, set_curdoc

curdoc().template =  jinja2.Template(source='''
	<!DOCTYPE html>
	<html lang="en">
    <head>
        <meta charset="utf-8">
        <title>{{ title if title else "Bokeh Plot" }}</title>
        {{ bokeh_css }}
        {{ bokeh_js }}
        <style>
        @import url(https://fonts.googleapis.com/css?family=Noto+Sans);
          body {
            width: 90%;
            height: 100%;
            margin: auto;
            text-align: justify;
            text-justify: inter-word;
            font-family: 'Noto Sans', sans-serif;
            -webkit-font-smoothing: antialiased;
            text-rendering: optimizeLegibility;
          }
          .bk-widget input[name$="vard"] {
             min-width: 50px !important;
             width: 100px !important;

invard = TextInput(name = "vard", value="", title = "Vardas", width = 130)
curdoc().add_root(invard)

So by name tag in TextInput(name="vard") and ".bk-widget input[name$="vard"]" and  changing width in bk-widget input, i could make textinput width as i want, but it don't work for version 12.4 and 12.5.

**Comment by bryevdv** [comment]
@Orasporas Your CSS target is not the correct one (also your template string was incomplete, I have updated the code above). Everything in a Bokeh document is under `bk-root`. There is no `bk-widget`. If I change the above to:
```
.bk-root input[name$="vard"] {
    min-width: 50px;
    width: 500px;
}
```
Then the text input shows up as 500px wide.

**Comment by bryevdv** [comment]
Actually, re-opening this. The above code sample does not actually illustrate a real problem, but this one does:
```
import jinja2
from bokeh.layouts import row
from bokeh.models.widgets import TextInput, Paragraph, Div, Select
from bokeh.io import curdoc, set_curdoc

curdoc().template =  jinja2.Template(source='''
    <!DOCTYPE html>
    <html lang="en">
    <head>
        <meta charset="utf-8">
        <title>{{ title if title else "Bokeh Plot" }}</title>
        {{ bokeh_css }}
        {{ bokeh_js }}
        <style>
        @import url(https://fonts.googleapis.com/css?family=Noto+Sans);
          body {
            width: 90%;
            height: 100%;
            margin: auto;
            text-align: justify;
            text-justify: inter-word;
            font-family: 'Noto Sans', sans-serif;
            -webkit-font-smoothing: antialiased;
            text-rendering: optimizeLegibility;
          }
          .bk-root input[name$="vard"] {
             min-width: 50px;
             width: 470px;
           }
        </style>
    </head>
    <body>
    {{ plot_div|indent(8) }}
    {{ plot_script|indent(8) }}
    </body>
''')

invard = TextInput(name = "vard", value="", title = "Vardas", width=500)
invard2 = TextInput(name = "vard2", value="", title = "Vardas")
curdoc().add_root(row(invard, invard2))
```

Without the added CSS the second text input is 500px to the right (as it should be) but the text input itself is not that wide. Without the `width=500` the second text input overlaps the first. Guessing there is simply not all the necessary coordination flowing out of `width=500`. That seems to inform the overall layout, but does not inform the test input itself. Hopefully a simple fix, but as there is a workaround for widgets in a `row` (the code above) I am pushing this to `0.12.7`

**Comment by Orasporas** [comment]
Nice, thanks.

**Comment by Orasporas** [comment]
Nice, thanks.

**Comment by bryevdv** [comment]
@mattpap this is fixed by #8085, the above code now yields

<img width="854" alt="screen shot 2019-01-30 at 21 11 16" src="https://user-images.githubusercontent.com/1078448/52031923-a10a0a80-24d3-11e9-9dba-8662c3ac7b28.png">
